src/gowrapper/ckks/ciphertext.go

```diff
@@ -42,20 +55,100 @@ func lattigo_ciphertextDegree(ctHandle Handle8) uint64 {
 
 //export lattigo_copyNew
 func lattigo_copyNew(ctHandle Handle8) Handle8 {
-	var ctIn *ckks.Ciphertext
+	var ctIn *rlwe.Ciphertext
 	ctIn = getStoredCiphertext(ctHandle)
 
-	var ctClone *ckks.Ciphertext
+	var ctClone *rlwe.Ciphertext
 	ctClone = ctIn.CopyNew()
 	return marshal.CrossLangObjMap.Add(unsafe.Pointer(ctClone))
 }
 
+//export lattigo_copyNewCiphertextQP
+func lattigo_copyNewCiphertextQP(ctQPHandle Handle8) Handle8 {
+	ctQPIn := getStoredCiphertextQP(ctQPHandle)
+	ctQPOut := ctQPIn.CopyNew()
+	return marshal.CrossLangObjMap.Add(unsafe.Pointer(ctQPOut))
+}
+
 //export lattigo_newCiphertext
-func lattigo_newCiphertext(paramsHandle Handle8, degree uint64, level uint64, scale float64) Handle8 {
+func lattigo_newCiphertext(paramsHandle Handle8, degree uint64, level uint64) Handle8 {
 	var params *ckks.Parameters
 	params = getStoredParameters(paramsHandle)
 
-	var newCt *ckks.Ciphertext
-	newCt = ckks.NewCiphertext(*params, int(degree), int(level), scale)
+	var newCt *rlwe.Ciphertext
+	newCt = ckks.NewCiphertext(*params, int(degree), int(level))
 	return marshal.CrossLangObjMap.Add(unsafe.Pointer(newCt))
 }
+
+//export lattigo_newZeroCiphertextQP
+func lattigo_newZeroCiphertextQP(paramsHandle, skHandle Handle8) Handle8 {
+	params := getStoredParameters(paramsHandle)
+	sk := getStoredSecretKey(skHandle)
+	ringQP := params.RingQP()
+	ctxQP := rlwe.CiphertextQP{Value: [2]ringqp.Poly{ringQP.NewPoly(), ringQP.NewPoly()}, MetaData: rlwe.MetaData{Scale: params.DefaultScale(), IsNTT: true}}
+	encryptor := ckks.NewEncryptor(*params, sk)
+	encryptor.EncryptZero(&ctxQP)
+	return marshal.CrossLangObjMap.Add(unsafe.Pointer(&ctxQP))
+}
+
+//export lattigo_setCiphertextMetaData
+func lattigo_setCiphertextMetaData(ctxHandle, metaDataHandle Handle8) {
+	ctx := getStoredCiphertext(ctxHandle)
+	metaData := getStoredMetaData(metaDataHandle)
+	(*ctx).MetaData = *metaData
+}
+
+//export lattigo_setCiphertextQPMetaData
+func lattigo_setCiphertextQPMetaData(ctxQPHandle, metaDataHandle Handle8) {
+	ctxQP := getStoredCiphertextQP(ctxQPHandle)
+	metaData := getStoredMetaData(metaDataHandle)
+	(*ctxQP).MetaData = *metaData
+}
+
+//export lattigo_getCiphertextMetaData
+func lattigo_getCiphertextMetaData(ctxHandle Handle8) Handle8 {
+	ctx := getStoredCiphertext(ctxHandle)
+	return marshal.CrossLangObjMap.Add(unsafe.Pointer(&ctx.MetaData))
+}
+
+//export lattigo_getCiphertextQPMetaData
+func lattigo_getCiphertextQPMetaData(ctxHandle Handle8) Handle8 {
+	ctxQP := getStoredCiphertextQP(ctxHandle)
+	return marshal.CrossLangObjMap.Add(unsafe.Pointer(&ctxQP.MetaData))
+}
+
+//export lattigo_poly
+func lattigo_poly(ctxHandle Handle8, i uint64) Handle8 {
```

Comment:
maybe change name to lattigo_ciphertextPoly to be consistent with the other method `lattigo_plaintextPoly` you added